spec/integration/primitives_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'spec_helper'
+require 'time'
+
+describe "chef-provisioning-docker" do
+  extend DockerSupport
+
+  when_the_chef_12_server "exists", organization: "foo", server_scope: :context do
+    with_docker "integration tests" do
+
+      ubuntu_options = {
+          :base_image => {
+            :name => 'ubuntu',
+            :repository => 'ubuntu',
+            :tag => '14.04'
+          },
+        }
+
+      let(:iso_date) { Time.now.iso8601.gsub(':', '') }
+      docker_driver = driver
+
+      context "machine_image resource" do
+
+          it ":create succeeds" do
+            spec_image_tag = "docker_image_spec_#{iso_date}"
+
+            expect_converge {
+              with_driver 'docker'
```

Comment:
Test fails without this, so `with_docker` is providing access to `driver`, but not making Docker the default driver for the enclosed contexts.